dango/dex/src/core/liquidity_pool.rs

```diff
@@ -93,6 +102,49 @@ pub trait PassiveLiquidityPool {
         reserve: CoinPair,
         output: Coin,
     ) -> anyhow::Result<(CoinPair, Coin)>;
+
+    /// Reflect the curve onto the orderbook.
+    ///
+    /// ## Inputs
+    ///
+    /// - `base_denom`: The base asset of the pool.
+    /// - `quote_denom`: The quote asset of the pool.
+    /// - `reserves`: The current pool reserves.
+    /// - `spread`: The spread between the ask and bid.
+    ///
+    /// ## Outputs
+    ///
+    /// - A tuple of two vectors of orders to place on the orderbook. The first vector
+    ///   contains the bids, the second contains the asks, specified as a tuple of
+    ///   (price, amount).
+    fn reflect_curve(
+        &self,
+        base_denom: Denom,
+        quote_denom: Denom,
+        reserves: &CoinPair,
+    ) -> StdResult<(
+        impl Iterator<Item = StdResult<(Udec128, Uint128)>>,
+        impl Iterator<Item = StdResult<(Udec128, Uint128)>>,
+    )>;
+
+    /// Returns the spot price of the pool at the current reserves, given as the number
+    /// of quote asset units per base asset unit.
+    ///
+    /// ## Inputs
+    ///
+    /// - `base_denom`: The base asset of the pool.
+    /// - `quote_denom`: The quote asset of the pool.
+    /// - `reserves`: The current pool reserves.
+    ///
+    /// ## Outputs
+    ///
+    /// - The spot price of the pool.
+    fn spot_price(
+        &self,
+        base_denom: &Denom,
+        quote_denom: &Denom,
+        reserves: &CoinPair,
+    ) -> StdResult<Udec128>;
```

Comment:
I realized in academic literature, this is called "marginal price". I've been using this term throughout the docs. See: https://github.com/left-curve/left-curve/tree/main/book/dex/passive-liquidity#order-placement